src/Frames/CSV.hs

```diff
@@ -46,51 +46,108 @@ import Control.Monad (void)
 defaultParser :: ParserOptions
-defaultParser = ParserOptions Nothing (T.pack defaultSep)
+defaultParser = ParserOptions Nothing defaultSep (RFC4180Quoting '\"')
 
 -- | Default separator string.
-defaultSep :: String
-defaultSep = ","
+defaultSep :: Separator
+defaultSep = T.pack ","
 
 -- * Parsing
 
 -- | Helper to split a 'T.Text' on commas and strip leading and
 -- trailing whitespace from each resulting chunk.
-tokenizeRow :: Separator -> T.Text -> [T.Text]
-tokenizeRow sep = map (unquote . T.strip) . T.splitOn sep
-  where unquote txt
-          | quoted txt = case T.dropEnd 1 (T.drop 1 txt) of
-                           txt' | T.null txt' -> "Col"
-                                | numish txt' -> txt
-                                | otherwise -> txt'
-          | otherwise = txt
-        numish = T.all (`elem` ("-+.0123456789"::String))
-        quoted txt = case T.uncons txt of
-                       Just ('"', rst)
-                         | not (T.null rst) -> T.last rst == '"'
-                       _ -> False
+tokenizeRow :: ParserOptions -> T.Text -> [T.Text]
+tokenizeRow options =
+    handleQuoting . T.splitOn sep
+  where sep = columnSeparator options
+        quoting = quotingMode options
+        handleQuoting = case quoting of
+          NoQuoting -> id
+          RFC4180Quoting quote -> reassembleRFC4180QuotedParts sep quote
+
+-- | Post processing applied to a list of tokens split by the
+-- separator which should have quoted sections reassembeld
+reassembleRFC4180QuotedParts :: Separator -> QuoteChar -> [T.Text] -> [T.Text]
+reassembleRFC4180QuotedParts sep quoteChar = finish . foldr f ([], Nothing)
+  where f :: T.Text -> ([T.Text], Maybe T.Text) -> ([T.Text], Maybe T.Text)
+        f part (rest, Just accum)
+          | prefixQuoted part = let token = unescape (T.drop 1 part) <> sep <> accum
+                                in (token : rest, Nothing)
+          | otherwise         = (rest, Just (unescape part <> sep <> accum))
+        f part (rest, Nothing)
+          | prefixQuoted part &&
+            suffixQuoted part = ((unescape . T.drop 1 . T.dropEnd 1 $ part) : rest, Nothing)
+          | suffixQuoted part = (rest, Just (unescape . T.dropEnd 1 $ part))
+          | otherwise         = (part : rest, Nothing)
+
+        prefixQuoted t =
+          quoteText `T.isPrefixOf` t &&
+          (T.length t - (T.length . T.dropWhile (== quoteChar) $ t)) `mod` 2 == 1
+        suffixQuoted t =
+          quoteText `T.isSuffixOf` t &&
+          (T.length t - (T.length . T.dropWhileEnd (== quoteChar) $ t)) `mod` 2 == 1
+
+        quoteText = T.singleton quoteChar
+
+        unescape :: T.Text -> T.Text
+        unescape = T.replace (quoteText <> quoteText) quoteText
+
+        finish :: ([T.Text], Maybe T.Text) -> [T.Text]
+        finish (rest, Just dangling) = dangling : rest -- FIXME? just assumes the close quote if it's missing
+        finish (rest, Nothing      ) =            rest
+
```

Comment:
Re Preface: Thanks!

yes, I missed the strip step. RFC4180 specifies no spaces between the comma and the quote, but I could imagine that people might want different. I'd think that stripping non-quoted tokens and preserving spaces inside quotes is noncontroversial though. We could also have some kind of `ParserOption` to turn on stripping inside quotes?

I used foldr just to avoid reversing the list at the end in finish, no particular laziness intent.

and yeah, definitely has several corner cases and should have test cases to go with them